### PR TITLE
OBGM-453 / OBGM-411 Unable to go forward/backward in outbound returns and product searches not working 

### PR DESCRIPTION
**Pull request by alannadolny**

### CODE REVIEW

**Comment by jmiranda** [comment]
This one needs to be rebased.

File: grails-app/domain/org/pih/warehouse/picklist/PicklistItem.groovy [hunk 1]
```diff
@@ -15,11 +15,16 @@ import org.pih.warehouse.inventory.InventoryItem
...
+import org.springframework.context.ApplicationEventPublisher
 
 class PicklistItem implements Serializable {
 
+    @Autowired
+    ApplicationEventPublisher applicationEventPublisher
+
     def publishRefreshEvent = {
-        Holders.grailsApplication.mainContext.publishEvent(new RefreshProductAvailabilityEvent(this))
+        applicationEventPublisher?.publishEvent(new RefreshProductAvailabilityEvent(this))
```
Comment (kchelstowski) [inline grails-app/domain/org/pih/warehouse/picklist/PicklistItem.groovy:27]: I think this refactor was not needed. Was it?

File: src/js/utils/apiClient.jsx [hunk 2]
```diff
@@ -9,13 +9,13 @@ import notification from 'components/Layout/notifications/notification';
 import LoginModal from 'components/LoginModal';
 import NotificationType from 'consts/notificationTypes';
 
-const justRejectRequestError = error => Promise.reject(error);
+const justRejectRequestError = (error) => Promise.reject(error);
```
Comment (kchelstowski) [inline src/js/utils/apiClient.jsx:12]: seems like ESLint rules have changed that he shouted for this? Or have you done it manually?

Comment (alannadolny) [reply src/js/utils/apiClient.jsx:12]: Yes, I used fix all eslint problems when I was working on this file and it added those brackets.

File: grails-app/domain/org/pih/warehouse/picklist/PicklistItem.groovy (hunk 1 above)
Comment (alannadolny) [reply grails-app/domain/org/pih/warehouse/picklist/PicklistItem.groovy]: You are right. Previously, without those changes, I got an error about it. For now, I reverted those changes and it works fine, thanks.

**Review by kchelstowski — APPROVED** [review]

**Review by awalkowiak — APPROVED** [review]

**Pull request merged**